server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -105,303 +105,305 @@ class AddressIndexRepository @Inject()(
           field = "paf.subBuildingName",
           value = token
-        ).boost(queryParams.subBuildingName.pafSubBuildingNameBoost)),
+        )).boost(queryParams.subBuildingName.pafSubBuildingNameBoost)),
       tokens.get(Tokens.subBuildingName).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoText",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoTextBoost)),
+        )).boost(queryParams.subBuildingName.lpiSaoTextBoost)),
       tokens.get(Tokens.saoStartSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.saoStartSuffix",
           value = token
-        ).boost(queryParams.subBuildingName.lpiSaoTextBoost))
+        )).boost(queryParams.subBuildingName.lpiSaoStartSuffixBoost))
     ).flatten
 
     val paoQuery = Seq(
       tokens.get(Tokens.paoStartNumber).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoStartNumber",
           value = token
-        ).boost(queryParams.buildingName.lpiPaoStartNumberBoost)),
+        )).boost(queryParams.buildingName.lpiPaoStartNumberBoost)),
       tokens.get(Tokens.paoStartSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoStartSuffix",
           value = token
-        ).boost(queryParams.buildingName.lpiPaoStartSuffixBoost)),
+        )).boost(queryParams.buildingName.lpiPaoStartSuffixBoost)),
       tokens.get(Tokens.paoEndNumber).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoEndNumber",
           value = token
-        ).boost(queryParams.buildingName.lpiPaoEndNumberBoost)),
+        )).boost(queryParams.buildingName.lpiPaoEndNumberBoost)),
       tokens.get(Tokens.paoEndSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoEndSuffix",
           value = token
-        ).boost(queryParams.buildingName.lpiPaoEndSuffixBoost))
+        )).boost(queryParams.buildingName.lpiPaoEndSuffixBoost))
     ).flatten
 
     val buildingNameQuery = Seq(
       tokens.get(Tokens.buildingName).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "paf.buildingName",
           value = token
-        ).boost(queryParams.buildingName.pafBuildingNameBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.buildingName.pafBuildingNameBoost)),
       tokens.get(Tokens.buildingName).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoText",
           value = token
-        ).boost(queryParams.buildingName.lpiPaoTextBoost)),
+        ).fuzziness(defaultFuzziness)).boost(queryParams.buildingName.lpiPaoTextBoost)),
       tokens.get(Tokens.paoStartSuffix).map(token =>
-        matchQuery(
+        constantScoreQuery(matchQuery(
           field = "lpi.paoStartSuffix",
```

Comment:
Same comments as in case of subBuildingName. BuildingName could be '66A-68D' and then PAO_START_NUMBER=66, PAO_START_SUFFIX=A, PAO_END_NUMBER=68, and PAO_END_SUFFIX=D. Perhaps we need to revisit this and the above when we add the parser post processing to extract the numbers and suffixes from these type of ranges.